File: java/src/main/java/de/tschuehly/microstreamtest/persistence/Root.java
package de.tschuehly.microstreamtest.persistence;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicLong;
import one.microstream.integrations.spring.boot.types.Storage;
import one.microstream.storage.types.StorageManager;
import org.springframework.beans.factory.annotation.Autowired;

@Storage
public class Root {
  @Autowired
  private StorageManager storageManager;

  private final AtomicLong index = new AtomicLong(0);

  Map<Long, Person> persons = new HashMap<>();
  public Long newIndex(){
    var newIndex = index.incrementAndGet();
    storageManager.store(index);
    return newIndex;
  }

  public  <T> void store(Map<Long,T> map){
    storageManager.store(map);
  }

}

File: src/main/kotlin/de/tschuehly/microstreamtest/persistence/Root.kt
package de.tschuehly.microstreamtest.persistence

import one.microstream.integrations.spring.boot.types.Storage
import one.microstream.storage.types.StorageManager
import org.springframework.beans.factory.annotation.Autowired
import java.util.concurrent.atomic.AtomicLong

@Storage
class Root {
    @Autowired
    @Transient
    private lateinit var storageManager: StorageManager
    private val index: AtomicLong = AtomicLong(0)

    val personMap = mutableMapOf<Long, Person>()

    fun newIndex(): Long {
        val newIndex = index.incrementAndGet()
        storageManager.store(index)
        return newIndex
    }

    fun <T> store(map: MutableMap<Long, T>) {
        storageManager.store(map)
    }
}

